src/include/ikarus/solver/nonLinearSolver/newtonRaphsonWithScalarSubsidiaryFunction.hh

```diff
@@ -0,0 +1,211 @@
+      /// It is assumed that Fext = Fext0 * lambda such that dRdlambda = Fext0
+      /// Generalization for Fext0 = Fext0(lambda) is not implemented
+
+      auto lambdaDummy = lambda;
+      auto DDummy      = x;
+      x.setZero();
+      lambda = 1.0;
+      nonLinearOperator().template update<0>();
+      const auto Fext0 = (-nonLinearOperator().value()).eval();  // dRdlambda(lambda)
+      lambda           = lambdaDummy;
+      x                = DDummy;
+
+      Eigen::MatrixX<double> residual2d, sol2d;
+      nonLinearOperator().updateAll();
+      const auto& rx = nonLinearOperator().value();
+      const auto& Ax = nonLinearOperator().derivative();
+
+      Eigen::VectorXd deltaD;
+      deltaD.resizeLike(rx);
+      deltaD.setZero();
+      double deltalambda = 0.0;
+
+      subsidiaryArgs.dfdDD.resizeLike(Fext0);
+      subsidiaryArgs.dfdDD.setZero();
+
+      subsidiaryFunction(subsidiaryArgs);
+      auto rNorm = sqrt(rx.dot(rx));
+      decltype(rNorm) dNorm;
+      int iter{0};
+      if constexpr (isLinearSolver) linearSolver.analyzePattern(Ax);
+
+      /// Iterative solving scheme
+      while (rNorm > settings.tol && iter < settings.maxIter) {
+        this->notify(NonLinearSolverMessages::ITERATION_STARTED);
+
+        /// Two-step solving procedure
+        residual2d.resize(rx.rows(), 2);
+        sol2d.resize(rx.rows(), 2);
+        sol2d.setZero();
+        residual2d.col(0) = -rx;
+        residual2d.col(1) = Fext0;
+
+        if constexpr (isLinearSolver) {
+          linearSolver.factorize(Ax);
+          linearSolver.solve(sol2d, residual2d);
+        } else {
+          sol2d = linearSolver(residual2d, Ax);
+        }
+
+        subsidiaryFunction(subsidiaryArgs);
+        this->notify(NonLinearSolverMessages::SCALARSUBSIDIARY_UPDATED, subsidiaryArgs.f);
+
+        deltalambda = (-subsidiaryArgs.f - subsidiaryArgs.dfdDD.dot(sol2d.col(0)))
+                      / (subsidiaryArgs.dfdDD.dot(sol2d.col(1)) + subsidiaryArgs.dfdDlambda);
+        deltaD = sol2d.col(0) + deltalambda * sol2d.col(1);
+
+        updateFunction(x, deltaD);
+        updateFunction(subsidiaryArgs.DD, deltaD);
+
+        lambda += deltalambda;
+        subsidiaryArgs.Dlambda += deltalambda;
+
+        dNorm = sqrt(deltaD.dot(deltaD) + deltalambda * deltalambda);
+        nonLinearOperator().updateAll();
+        rNorm = sqrt(rx.dot(rx) + subsidiaryArgs.f * subsidiaryArgs.f);
+
+        this->notify(NonLinearSolverMessages::SOLUTION_CHANGED);
+        this->notify(NonLinearSolverMessages::CORRECTIONNORM_UPDATED, dNorm);
+        this->notify(NonLinearSolverMessages::RESIDUALNORM_UPDATED, rNorm);
+        this->notify(NonLinearSolverMessages::ITERATION_ENDED);
+        ++iter;
+      }
+
+      if (iter == settings.maxIter) solverInformation.success = false;
+      solverInformation.iterations   = iter;
+      solverInformation.residualnorm = rNorm;
+      this->notify(NonLinearSolverMessages::FINISHED_SUCESSFULLY, iter, rNorm, settings.tol);
```

Comment:
If in line 174 we determine that we have failed, then we should not notify FINISHED_SUCESSFULLY here. Maybe this is also wrong in the normal NewtoRhapson?